redash/settings/__init__.py

```diff
@@ -48,9 +49,77 @@ def all_settings():
 SCHEMAS_REFRESH_SCHEDULE = int(os.environ.get("REDASH_SCHEMAS_REFRESH_SCHEDULE", 30))
 
 AUTH_TYPE = os.environ.get("REDASH_AUTH_TYPE", "api_key")
-ENFORCE_HTTPS = parse_boolean(os.environ.get("REDASH_ENFORCE_HTTPS", "false"))
 INVITATION_TOKEN_MAX_AGE = int(os.environ.get("REDASH_INVITATION_TOKEN_MAX_AGE", 60 * 60 * 24 * 7))
 
+# Whether and how to redirect non-HTTP requests to HTTPS. Disabled by default.
+ENFORCE_HTTPS = parse_boolean(os.environ.get("REDASH_ENFORCE_HTTPS", "false"))
+ENFORCE_HTTPS_PERMANENT = parse_boolean(
+    os.environ.get("REDASH_ENFORCE_HTTPS_PERMANENT", "false"))
+# Whether file downloads are enforced or not.
+ENFORCE_FILE_SAVE = parse_boolean(
+    os.environ.get("REDASH_ENFORCE_FILE_SAVE", "true"))
+
+COOKIE_SECRET = os.environ.get(
+    "REDASH_COOKIE_SECRET", "c292a0a3aa32397cdb050e233733900f")
+# Whether the session cookie is set to secure.
+SESSION_COOKIE_SECURE = parse_boolean(
+    os.environ.get("REDASH_SESSION_COOKIE_SECURE") or str(ENFORCE_HTTPS))
+# Whether the session cookie is set HttpOnly.
+SESSION_COOKIE_HTTPONLY = parse_boolean(
+    os.environ.get("REDASH_SESSION_COOKIE_HTTPONLY", "true"))
+
+# Doesn't set X-Frame-Options by default since it's highly dependent
+# on the specific deployment.
+# See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options
+# for more information.
+FRAME_OPTIONS = os.environ.get("REDASH_FRAME_OPTIONS", None)
```

Comment:
Does Talisman support setting this on a per route basis? Because we should set this by default to `deny` for all routes except the embeddable ones.

Reply:
@arikfr Yep, it does, like I did for the [ping endpoint](https://github.com/getredash/redash/pull/3404/files#diff-5935f49d3a59e4651b814a952cd84919R12). More info: https://github.com/GoogleCloudPlatform/flask-talisman#per-view-options

Reply:
Then how about we use a strict policy here by default, and use per-view options to allow embedding for: visualization embed, public dashboard and non public dashboard?

Reply:
That works for me, I'll append it..

Reply:
Thanks.

Reply:
Wait, we have a problem, according to the spec:

```
   Wildcards or lists to declare multiple domains in one ALLOW-FROM
   statement are not permitted (see Section 2.3.2.3).
```

So we can't specify a list of URLs or a wildcard for the embed endpoints when using `X-Frame-Options: allow-from https://example.com/`. I don't think it's practical for us to spell out the individual URLs that embedding is allowed either.

There is a better alternative though, the [CSP directive frame-ancestors](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-ancestors). It's [in CSP 2](https://www.w3.org/TR/CSP2/#directive-frame-ancestors), which means we should simply use the CSP header. Browser support for it isn't great though, IE and Mobile Edge don't support it, while X-Frame-Options has broader support. My gut says we should go with it anyway  though. What do you think?

Reply:
I thought of allowing a wildcard there.